digits/model/tasks/caffe_train.py

```diff
@@ -1049,61 +1057,15 @@ def can_infer_one(self):
         return False
 
     @override
-    def infer_one(self, data, snapshot_epoch=None, layers=None):
-        if isinstance(self.dataset, dataset.ImageClassificationDatasetJob):
-            return self.classify_one(data,
-                    snapshot_epoch=snapshot_epoch,
-                    layers=layers,
-                    )
-        elif isinstance(self.dataset, dataset.GenericImageDatasetJob):
-            return self.infer_one_generic(data,
+    def infer_one(self, data, snapshot_epoch=None, layers=None, gpu=-1):
+        if isinstance(self.dataset, dataset.ImageClassificationDatasetJob) or isinstance(self.dataset, dataset.GenericImageDatasetJob):
+            return self.infer_one_image(data,
                     snapshot_epoch=snapshot_epoch,
                     layers=layers,
```

Comment:
You're missing a `gpu=gpu` here which is causing you to never call `set_mode_gpu()`.